stackslib/src/net/api/tests/getclaritymetadata.rs

```diff
@@ -0,0 +1,273 @@
+        )
+        .unwrap();
+
+    // parsed request consumes headers that would not be in a constructed request
+    parsed_request.clear_headers();
+    let (preamble, contents) = parsed_request.destruct();
+
+    // consumed path args
+    assert_eq!(handler.clarity_metadata_key, Some("analysis".to_string()));
+    assert_eq!(
+        handler.contract_identifier,
+        Some(
+            QualifiedContractIdentifier::parse(
+                "ST2DS4MSWSGJ3W9FBC6BVT0Y92S345HY8N3T6AV7R.hello-world"
+            )
+            .unwrap()
+        )
+    );
+
+    assert_eq!(&preamble, request.preamble());
+
+    handler.restart();
+    assert!(handler.clarity_metadata_key.is_none());
+}
+
+#[test]
+fn test_try_make_response() {
+    let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 33333);
+
+    let mut requests = vec![];
+
+    // query existing contract size metadata
+    let request = StacksHttpRequest::new_getclaritymetadata(
+        addr.into(),
+        StacksAddress::from_string("ST2DS4MSWSGJ3W9FBC6BVT0Y92S345HY8N3T6AV7R").unwrap(),
+        "hello-world".try_into().unwrap(),
+        "vm-metadata::9::contract-size".to_string(),
+        TipRequest::UseLatestAnchoredTip,
+    );
+    requests.push(request);
+
+    // query existing data var metadata
+    let request = StacksHttpRequest::new_getclaritymetadata(
+        addr.into(),
+        StacksAddress::from_string("ST2DS4MSWSGJ3W9FBC6BVT0Y92S345HY8N3T6AV7R").unwrap(),
+        "hello-world".try_into().unwrap(),
+        "vm-metadata::5::test-map".to_string(),
+        TipRequest::UseLatestAnchoredTip,
+    );
+    requests.push(request);
+
+    // query existing data map metadata
+    let request = StacksHttpRequest::new_getclaritymetadata(
+        addr.into(),
+        StacksAddress::from_string("ST2DS4MSWSGJ3W9FBC6BVT0Y92S345HY8N3T6AV7R").unwrap(),
+        "hello-world".try_into().unwrap(),
+        "vm-metadata::6::bar".to_string(),
+        TipRequest::UseLatestAnchoredTip,
+    );
+    requests.push(request);
+
+    let mut responses = test_rpc(function_name!(), requests);
+
+    // contract size metadata
+    let response = responses.remove(0);
+    assert_eq!(
+        response.preamble().get_canonical_stacks_tip_height(),
+        Some(1)
+    );
+    let resp = response.decode_clarity_metadata_response().unwrap();
+    assert_eq!(resp.data, "1432");
+
+    // data map metadata
+    let response = responses.remove(0);
+    let resp = response.decode_clarity_metadata_response().unwrap();
+    let expected = DataMapMetadata {
+        key_type: TypeSignature::UIntType,
+        value_type: TypeSignature::UIntType,
+    };
+    assert_eq!(resp.data, expected.serialize());
+
+    // data var metadata
+    let response = responses.remove(0);
+    let resp = response.decode_clarity_metadata_response().unwrap();
+    let expected = DataVariableMetadata {
+        value_type: TypeSignature::IntType,
+    };
+    assert_eq!(resp.data, expected.serialize());
```

Comment:
Can you add test coverage here for HTTP 400 and HTTP 404 error cases?

Reply:
Done